core/trino-main/src/test/java/io/trino/operator/output/TestPartitionedOutputOperator.java

```diff
@@ -13,248 +13,693 @@
+        assertThat(partition1).containsExactly(0L); // position 0 copied to all partitions
+    }
+
+    @Test
+    public void testPartitionPositionsWithRleNullWithNullChannel()
+    {
+        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT, BIGINT).withNullChannel(0).build();
+        Page page = new Page(new RunLengthEncodedBlock(createLongsBlock((Long) null), POSITIONS_PER_PAGE), createLongSequenceBlock(0, POSITIONS_PER_PAGE));
+
+        processPages(partitionedOutputOperator, page);
+
+        List<Object> partition0 = readLongs(outputBuffer.getEnqueuedDeserialized(0), 1);
+        assertThat(partition0).containsExactlyElementsOf(readLongs(Stream.of(page), 1));
+        List<Object> partition1 = readLongs(outputBuffer.getEnqueuedDeserialized(1), 1);
+        assertThat(partition1).containsExactlyElementsOf(readLongs(Stream.of(page), 1));
+    }
+
+    @Test
+    public void testOutputForDictionaryBlock()
+    {
+        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        Page page = new Page(createLongDictionaryBlock(0, 10)); // must have at least 10 position to have non-trivial dict
+
+        processPages(partitionedOutputOperator, page);
+
+        List<Object> partition0 = readLongs(outputBuffer.getEnqueuedDeserialized(0), 0);
+        assertThat(partition0).containsExactlyElementsOf(nCopies(5, 0L));
+        List<Object> partition1 = readLongs(outputBuffer.getEnqueuedDeserialized(1), 0);
+        assertThat(partition1).containsExactlyElementsOf(nCopies(5, 1L));
+    }
+
+    @Test
+    public void testOutputForOneValueDictionaryBlock()
+    {
+        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        Page page = new Page(new DictionaryBlock(createLongsBlock(0), new int[] {0, 0, 0, 0}));
+
+        processPages(partitionedOutputOperator, page);
+
+        List<Object> partition0 = readLongs(outputBuffer.getEnqueuedDeserialized(0), 0);
+        assertThat(partition0).containsExactlyElementsOf(nCopies(4, 0L));
+        List<Object> partition1 = readLongs(outputBuffer.getEnqueuedDeserialized(1), 0);
+        assertThat(partition1).isEmpty();
+    }
+
+    @Test
+    public void testOutputForViewDictionaryBlock()
+    {
+        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        Page page = new Page(new DictionaryBlock(createLongSequenceBlock(4, 8), new int[] {1, 0, 3, 2}));
+
+        processPages(partitionedOutputOperator, page);
+
+        List<Object> partition0 = readLongs(outputBuffer.getEnqueuedDeserialized(0), 0);
+        assertThat(partition0).containsExactlyInAnyOrder(4L, 6L);
+        List<Object> partition1 = readLongs(outputBuffer.getEnqueuedDeserialized(1), 0);
+        assertThat(partition1).containsExactlyInAnyOrder(5L, 7L);
+    }
+
+    @DataProvider(name = "types")
+    public static Object[][] types()
```

Comment:
Move that below `testOutputForSimplePageWithType`

Reply:
done